pkg/exec/exec.go

```diff
@@ -124,6 +126,29 @@ func (e OSExecutor) CreateInstance(ctx context.Context, imageID int, instanceID
 	return err
 }
 
+// RetrieveInstanceCredentials reads the certificate and key files from the
+// instance directory and returns them in a map
+func (e OSExecutor) RetrieveInstanceCredentials(ctx context.Context, id int) (map[string][]byte, error) {
```

Comment:
Unsure of the answer here: do we need the context?

Reply:
We do unfortunately, as that's how we get a reference to the logger in the functions in this package.

Reply:
Sigh. Not the end of the world though.